### PR TITLE
feat: associations

### PR DESCRIPTION
**Pull request by albertyfwu**
## Test Plan

[Describe test plan here]

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: apps/api/routes/crm/v2/association_type/index.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      const { source_object_class_id, source_object_class_type, target_object_class_id, target_object_class_type } =
+        req.query;
+
+      if (
+        !source_object_class_id ||
+        !source_object_class_type ||
+        !target_object_class_id ||
+        !target_object_class_type
+      ) {
+        throw new Error('Missing required query params');
```
Comment (asdfryan) [inline apps/api/routes/crm/v2/association_type/index.ts:41]: BadRequestError

File: packages/core/remotes/crm/hubspot/index.ts [hunk 2]
```diff
@@ -869,13 +872,12 @@ class HubSpotClient extends AbstractCrmRemoteClient {
...
     await this.maybeRefreshAccessToken();
     const response = await this.#client.crm.schemas.coreApi.getById(id);
-
     return {
       id: response.objectTypeId,
       name: response.name,
       description: null,
       labels: {
-        // TODO
+        // TODO:
```
Comment (asdfryan) [inline packages/core/remotes/crm/hubspot/index.ts:880]: ? Were you going to put a ticket number?

File: packages/core/remotes/crm/hubspot/index.ts [hunk 3]
```diff
@@ -1018,6 +1020,65 @@ class HubSpotClient extends AbstractCrmRemoteClient {
...
+      sourceObjectClass,
+      targetObjectClass,
+      displayName: result.label ?? '',
+      cardinality: 'UNKNOWN',
+    }));
+  }
+
+  public async createAssociationType(params: AssociationTypeCreateParams): Promise<void> {
+    if (params.cardinality !== 'ONE_TO_MANY') {
+      throw new Error('Only ONE_TO_MANY cardinality is supported in HubSpot');
```
Comment (asdfryan) [inline packages/core/remotes/crm/hubspot/index.ts:1043]: BadRequestError (to ensure we get 4xx errors)

File: packages/core/remotes/crm/hubspot/index.ts (hunk 2 above)
Comment (albertyfwu) [reply packages/core/remotes/crm/hubspot/index.ts:880]: no i'll get rid of this

File: apps/api/routes/crm/v2/association_type/index.ts (hunk 1 above)
Comment (albertyfwu) [reply apps/api/routes/crm/v2/association_type/index.ts:41]: ok

**Pull request merged**